text/accumulation.tex

```diff
@@ -30,118 +34,206 @@ \subsection{Preimage Integration}
 
 
 
-\subsection{Gas Accounting}
 
-We define $\mathbf{S}$, the set of all services which will be accumulated in this block; this is all services which have at least one work output within $\mathbf{W}$, together with the privileged services, $\chi_\mathbf{g}$. Formally:
-\begin{align}\label{eq:servicestoaccumulate}
-  \mathbf{S} \equiv \{ \mathbf{r}_s \mid w \in \mathbf{W}, \mathbf{r} \in w_\mathbf{r} \} \cup \keys(\chi_\mathbf{g})
-\end{align}
 
-We calculate the gas attributable for each service as the sum of each of the service's work outputs' share of their report's elective accumulation gas together with the subtotal of minimum gas requirements:
-\begin{equation}
-  G\colon\left\{\;\begin{aligned}
-    \N_S &\to \N_G \\
-    \displaystyle s &\mapsto \!\!\sum_{w \in \mathbf{W}}\sum_{\mathbf{r} \in w_\mathbf{r} , \mathbf{r}_s = s}{\!\!\!\!\!\!\!\delta^\dagger[s]_g + \left \lfloor \mathbf{r}_g\cdot\frac{\displaystyle \mathsf{G}_A - \sum_{\mathbf{r} \in w_\mathbf{r}}{\delta^\dagger[\mathbf{r}_s]_g}}{\sum_{\mathbf{r} \in w_\mathbf{r}}{\mathbf{r}_g}} \right \rfloor }
-  \end{aligned}\right.\!\!\!\!\!\!
-\end{equation}
+\subsection{History and Queuing}
 
-\subsection{Wrangling}
+Accumulation of a work-package/work-report is deferred in the case that it has a not-yet-fulfilled dependency and is cancelled entirely in the case of an invalid dependency. Dependencies are specified as work-package hashes and in order to know which work-packages have been accumulated already, we maintain a history of what has been accumulated. This history, $\accumulated$, is sufficiently large for an epoch worth of work-reports. For all such reports, we also retain their segment-root in order that we may validate any guarantor-proscribed segment-roots in work-reports prior to accumulation. Formally:
+\begin{align}
+  \accumulated &\in \seq{\dict{\H}{\H}}_\mathsf{E} \\
+  \wideparen{\accumulated} &\equiv \{ x \mid x \in \accumulated \}
+\end{align}
 
-We finally define the results which will be given as an operand into the accumulate function for each service in $\mathbf{S}$. This is a sequence of operand tuples $\mathbb{O}$, one sequence for each service in $\mathbf{S}$. Each sequence contains one element per work output (or error) to be accumulated for that service, together with said work output's payload hash, package hash and authorization output. The tuples are sequenced in the same order as they appear in $\mathbf{W}$. Formally:
+We also maintain knowledge of available but not-yet-accumulated work-reports in the state item $\ready$. Each of these have been available for at most one epoch but have or had unfulfilled dependencies. Alongside the work-report itself, we retain its unaccumulated dependencies, a set of work-package hashes. Formally:
```

Comment:
Not sure "available" is a good term here as the WP bundles might not be available any more -- the reports could have been audited, the blocks which queued them finalized, and the bundle shards discarded by assurers.